bubblewrap.c

```diff
@@ -1233,6 +1260,47 @@ setup_newroot (bool unshare_pid,
                          0, 0, source, dest);
           break;
 
+        case SETUP_OVERLAY_MOUNT:
+        case SETUP_RO_OVERLAY_MOUNT:
+        case SETUP_TMP_OVERLAY_MOUNT:
+          {
+            StringBuilder sb = {0};
+            bool multi_src = FALSE;
+
+            if (mkdir (dest, 0755) != 0 && errno != EEXIST)
+              die_with_error ("Can't mkdir %s", op->dest);
+
+            if (op->source != NULL)
+              {
+                strappend (&sb, "upperdir=/oldroot");
+                strappend_escape_for_mount_options (&sb, op->source);
+                strappend (&sb, ",workdir=/oldroot");
+                op = op->next;
+                strappend_escape_for_mount_options (&sb, op->source);
+                strappend (&sb, ",");
+              }
+            else if (op->type == SETUP_TMP_OVERLAY_MOUNT)
+              strappendf (&sb, "upperdir=/tmp-overlay-upper-%1$d,workdir=/tmp-overlay-work-%1$d,",
+                          tmp_overlay_idx++);
+
+            strappend (&sb, "lowerdir=/oldroot");
+            while (op->next != NULL && op->next->type == SETUP_OVERLAY_SRC)
+              {
+                op = op->next;
+                if (multi_src)
+                  strappend (&sb, ":/oldroot");
+                strappend_escape_for_mount_options (&sb, op->source);
+                multi_src = TRUE;
+              }
+
+            strappend (&sb, ",userxattr");
```

Comment:
FWIW Podman (https://github.com/containers/storage/blob/a0eeb0781129973c70d2cb101b00e312631a5702/drivers/overlay/overlay.go#L1723) does not add this flag if running as the real root (or rather: in the initial user namespace + has `CAP_SYS_ADMIN`). I don't think this matters here?

Reply:
Shouldn't; this whole feature is disabled when running setuid.

Reply:
What I had in mind is the case where bubblewrap is run as root directly (without being elevated through setuid), at first glance it looks like it's supported, there's even [someone using it](https://github.com/containers/bubblewrap/blob/8e51677abd7e3338e4952370bf7d902e37d8cbb6/bubblewrap.c#L880) and in this case `is_privileged = FALSE` so features which are disabled when running setuid, including overlay, still work.

Though IMO unless someone comes with a good reason why `userxattr` should not be used in this case, I'd rather just keep `userxattr` and not handle it specially, specially since checking for it is a bit of a PITA.